python/createCollectionDownload.py

```diff
@@ -2,75 +2,189 @@
 import requests
 import json
 import time
+from datetime import datetime
 
-def check_api_key(cxy_api_key):
+def request_api(url:str, headers:str, method='GET', params=None, data={}):
+    """
+    Handles making requests to the ChainXY API and returns the response.
+    """
+    r = requests.request(url=url, method=method, headers=headers, params=params, data=data)
+    if r.status_code == 401:
+        raise ValueError("Bad ChainXY API key provided, please double-check the input value!")
+    elif r.status_code != 200 and 'does not correspond to a collection' in r.text.lower():
+        print("The collection does not exist in the database.")
+        r = None
+    else:
+        return r
+
+def validate_inputs(headers:str, cxy_api_key:str, collection_type:str, cache_time:int):
+    """
+    Verifies that a given API key is valid for making requests to the API and checks that the collection download
+    inputs are valid for processing.
+    """
     url = 'https://location.chainxy.com/api/Users/Me'
-    headers = {'x-apikey': cxy_api_key,
-            'x-Application': 'Python API Call',
-            'content-type': 'application/json'}
-    response = requests.get(url, headers=headers)
-    if response.status_code == 401:
-        raise ValueError("Bad ChainXY API key provided, double-check the provided value!")
+    r = request_api(url=url, headers=headers)
+    if not isinstance(collection_type, str) or collection_type.lower() not in ['chain', 'center']:
+        raise ValueError(f"Collection type '{collection_type}' must be one of ['chain', 'center'].")
+    
+    if cache_time < 0:
+        raise ValueError("Please enter a valid duration for cache_time that is an int > 0.")
+
+def get_collection_urls(collection_id:str, collection_type:str):
```

Comment:
What's the benefit of this living as a separate function? It seems like it just sets the values conditionally.

Reply:
I was thinking to declutter `download_collection()` and have it just focus on the actions related to downloading a collection as the name suggests, but the assignments are readable in either case.

After thinking about it more, I don't think we'll ever use `get_collection_urls()` for any other purpose than in `download_collection()`, and it's pretty obvious where it is if we ever need to change it, so I'll revert this change.

Resolved!